sentry-android-core/src/main/java/io/sentry/android/core/ManifestMetadataReader.java

```diff
@@ -59,132 +61,113 @@ static void applyMetadata(
-        options.getLogger().log(SentryLevel.DEBUG, "enableAppLifecycleBreadcrumbs read: %s", ndk);
-        options.setEnableAppLifecycleBreadcrumbs(enableAppLifecycleBreadcrumbs);
-
-        final boolean enableSystemEventBreadcrumbs =
-            metadata.getBoolean(
-                BREADCRUMBS_SYSTEM_EVENTS_ENABLE, options.isEnableSystemEventBreadcrumbs());
-        options.getLogger().log(SentryLevel.DEBUG, "enableSystemEventBreadcrumbs read: %s", ndk);
-        options.setEnableSystemEventBreadcrumbs(enableSystemEventBreadcrumbs);
-
-        final boolean enableAppComponentBreadcrumbs =
-            metadata.getBoolean(
-                BREADCRUMBS_APP_COMPONENTS_ENABLE, options.isEnableAppComponentBreadcrumbs());
-        options.getLogger().log(SentryLevel.DEBUG, "enableAppComponentBreadcrumbs read: %s", ndk);
-        options.setEnableAppComponentBreadcrumbs(enableAppComponentBreadcrumbs);
-
-        final boolean enableUncaughtExceptionHandler =
-            metadata.getBoolean(
-                UNCAUGHT_EXCEPTION_HANDLER_ENABLE, options.isEnableUncaughtExceptionHandler());
-        options.getLogger().log(SentryLevel.DEBUG, "enableUncaughtExceptionHandler read: %s", ndk);
-        options.setEnableUncaughtExceptionHandler(enableUncaughtExceptionHandler);
-
-        final boolean attachThreads =
-            metadata.getBoolean(ATTACH_THREADS, options.isAttachThreads());
-        options.getLogger().log(SentryLevel.DEBUG, "attachThreads read: %s", attachThreads);
-        options.setAttachThreads(attachThreads);
+                options.isEnableActivityLifecycleBreadcrumbs()));
+
+        options.setEnableAppLifecycleBreadcrumbs(
+            readBool(
+                metadata,
+                logger,
+                BREADCRUMBS_APP_LIFECYCLE_ENABLE,
+                options.isEnableAppComponentBreadcrumbs()));
+
+        options.setEnableSystemEventBreadcrumbs(
+            readBool(
+                metadata,
+                logger,
+                BREADCRUMBS_SYSTEM_EVENTS_ENABLE,
+                options.isEnableSystemEventBreadcrumbs()));
+
+        options.setEnableAppComponentBreadcrumbs(
+            readBool(
+                metadata,
+                logger,
+                BREADCRUMBS_APP_COMPONENTS_ENABLE,
+                options.isEnableAppComponentBreadcrumbs()));
+
+        options.setEnableUncaughtExceptionHandler(
+            readBool(
+                metadata,
+                logger,
+                UNCAUGHT_EXCEPTION_HANDLER_ENABLE,
+                options.isEnableUncaughtExceptionHandler()));
+
+        options.setAttachThreads(
+            readBool(metadata, logger, ATTACH_THREADS, options.isAttachThreads()));
+
+        if (options.getTracesSampleRate() == null) {
+          final Double tracesSampleRate = readDouble(metadata, logger, TRACES_SAMPLE_RATE);
+          if (tracesSampleRate != -1) {
+            options.setTracesSampleRate(tracesSampleRate);
+          }
+        }
```

Comment:
This is what needs review, right?

Reply:
yes, I was unhappy with all the boilerplate I've written before and every time was growing, even more, so added some extra methods to read/parse easily, also added tests to all the missing fields, increasing testability and readability